### ISSUE TITLE
Waiting for firefox to connect

### ISSUE THREAD

**Issue by hazcod** [browsh-org/browsh#121]
Hi,

I cannot use browsh, it stays at 'connecting to firefox'.
Browsh version: browsh 1.3.3
macOS version: 10.13.6
Firefox version: 61.0.1

I run it with:
`./browsh -firefox /Applications/Firefox.app/Contents/MacOS/firefox-bin -debug`

Debug log:
```
Attempting to connect to Firefox Marionette
Starting Firefox in headless mode
Using default profile at: /Users/xxxx/Library/Application Support/browsh/firefox_profile
FF-CONSOLE: 1531379373774	Marionette	INFO	Listening on port 2828
FF-MRNT: 50:{"applicationType":"gecko","marionetteProtocol":3}
Sending `newSession` to Firefox Marionette
FF-MRNT: 539:[1,0,null,{"sessionId":"094adf1f-4437-f840-a6be-80b196ac2243","capabilities":{"browserName":"firefox","browserVersion":"61.0.1","platformName":"darwin","platformVersion":"17.7.0","pageLoadStrategy":"normal","acceptInsecureCerts":false,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"rotatable":false,"moz:accessibilityChecks":false,"moz:headless":true,"moz:processID":78885,"moz:profile":"/Users/xxx/Library/Application Support/browsh/firefox_profile","moz:useNonSpecCompliantPointerOrigin":false,"moz:webdriverClick":true}}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 13:[1,1,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 25:[1,2,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 13:[1,3,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 13:[1,4,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 25:[1,5,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 13:[1,6,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 13:[1,7,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 25:[1,8,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 13:[1,9,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,10,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,11,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,12,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,13,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,14,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,15,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,16,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,17,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,18,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,19,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,20,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,21,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,22,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,23,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,24,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,25,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,26,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,27,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,28,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,29,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,30,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,31,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,32,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,33,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,34,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,35,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,36,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,37,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,38,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,39,null,{}]
Sending `addon:install` to Firefox Marionette
FF-MRNT: 62:[1,40,null,{"value":"{8ff2d753-2dc8-46de-a837-fa28331d9fcf}"}]
Starting Browsh CLI client
Webextension not connected. Message not sent: /tty_size,274,87
Incoming web request from browser
TTY sending: /tty_size,274,87
TTY sending: /new_tab,https://google.com
WEBEXT: Webextension connected to the terminal's websocket server
WEBEXT: Starting to listen to TTY
WEBEXT: BACKGROUND: Frame loop starting
WEBEXT: Message from terminal: /tty_size,274,87
WEBEXT: Requesting browser resize for new TTY dimensions: 274x87
WEBEXT: Not resizing browser window without all of the TTY and character dimensions
WEBEXT: Message from terminal: /new_tab,https://google.com
WEBEXT: Tab undefined received new focus
WEBEXT: New tab created: [object Object]
WEBEXT: Not resizing browser window without all of the TTY and character dimensions
WEBEXT: Active tab 1 does not have a channel, so not requesting a frame
WEBEXT: Active tab 1 does not have a channel, so not requesting a frame
WEBEXT: Active tab 1 does not have a channel, so not requesting a frame
WEBEXT: Active tab 1 does not have a channel, so not requesting a frame
WEBEXT: Active tab 1 does not have a channel, so not requesting a frame
WEBEXT: Active tab 1 does not have a channel, so not requesting a frame
WEBEXT: Active tab 1 does not have a channel, so not requesting a frame
WEBEXT: Active tab 1 does not have a channel, so not requesting a frame
...
```

**Comment by tombh** [comment]
From the logs I can see that Firefox does connect. But it's not loading the first tab. Curiously I most often see this when there's no Internet, or the Internet is slow. You might just need to wait a little longer.

Another thing you can try is closing all your Firefox processes. Then starting Browsh again.

**Comment by ThyrixYang** [comment]
This may because google is blocked in your area, google is the default home page.

**Comment by HazCod** [comment]
Can confirm firefox was not connecting through the usual http proxy

**Comment by greenled** [comment]
Hi. I'm also running the standalone version (in Debian) and behind a proxy, but adding --debug doesn't show any debug info. It just freezes at "Waiting for Firefox to connect". Also tried with a local url (one that needs no proxy) and got the same result. Should I put this in another issue?

Browsh version: browsh 1.4.12
debian version: 9 (Squeze)
Firefox version: 61.0.1

**Comment by greenled** [comment]
Sorry, logs were sent to ``./debug.log``:

```
Looking in /home/juan/.config/browsh/config.toml for config.
Attempting to connect to Firefox Marionette
Starting Firefox in headless mode
Using Firefox at: /opt/firefox/firefox/firefox
Using default profile at: /home/juan/.config/browsh/firefox_profile
FF-CONSOLE: 1533590144030	Marionette	INFO	Listening on port 2828
FF-MRNT: 50:{"applicationType":"gecko","marionetteProtocol":3}
Sending `newSession` to Firefox Marionette
FF-MRNT: 522:[1,0,null,{"sessionId":"07ac3870-77ee-41f5-9ec5-72ffdaf50d3c","capabilities":{"browserName":"firefox","browserVersion":"61.0.1","platformName":"linux","platformVersion":"4.15.18-1-pve","pageLoadStrategy":"normal","acceptInsecureCerts":false,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"rotatable":false,"moz:accessibilityChecks":false,"moz:headless":true,"moz:processID":7043,"moz:profile":"/home/juan/.config/browsh/firefox_profile","moz:useNonSpecCompliantPointerOrigin":false,"moz:webdriverClick":true}}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 13:[1,1,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 25:[1,2,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 13:[1,3,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 13:[1,4,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 25:[1,5,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 13:[1,6,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 13:[1,7,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 25:[1,8,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 13:[1,9,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,10,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,11,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,12,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,13,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,14,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,15,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,16,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,17,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,18,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,19,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,20,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,21,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,22,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,23,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,24,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,25,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,26,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,27,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,28,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,29,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,30,null,{}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,31,null,{}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,32,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 14:[1,33,null,{}]
Sending `addon:install` to Firefox Marionette
FF-MRNT: 62:[1,34,null,{"value":"{8ff2d753-2dc8-46de-a837-fa28331d9fcf}"}]
Starting Browsh CLI client
Webextension not connected. Message not sent: /tty_size,150,38
Webextension not connected. Message not sent: /tty_size,150,36
```

**Comment by tombh** [comment]
Hmm, like I mentioned before, the only time I've ever seen this is when Firefox can't connect to the internet. But I doubt this is the case for you?

**Comment by greenled** [comment]
Well, I am behind a corporate proxy, which makes connection a bit trickier. I already have set http_proxy and https_proxy environment variables in /etc/environment, and so far all commands I run can use internet without troubles. Does the launched Firefox instance respect those environment variables by default?

**Comment by JackieMium** [comment]
Same issue here. I've got a system-wide proxy on, yet bowsh still got stuck at

> Starting Browsh v1.4.12, the modern text-based web browser.
> Waiting for Fireefox to connect...

**Comment by mixn** [comment]
I can confirm this issue. I’m running `1.4.12`, am not behind a proxy and have Firefox.app open + can access `google.com`.

**Comment by greenled** [comment]
Update: tried inside a VPN connection and browsh works as expected

**Comment by 10by10pixel** [comment]
Was referred here from #213. Is there a solution found yet? I'm not on VPN/Proxy. I am able to access internet through Firefox.app.

**Comment by tombh** [comment]
What about trying different sites with `--startup-url`?

**Comment by 10by10pixel** [comment]
Doesn't seem to help. Tried the following:

- https://google.com
- http://google.com
- http://owenwalters.net/

**Comment by megdimka** [comment]
NixOS 18.09.1822.0a8f35c998f (Jellyfish)
Mozilla Firefox 64.0
Browsh 1.4.10
[debug.log](https://github.com/browsh-org/browsh/files/2720664/debug.log)

**Comment by younger-1** [comment]
I use browsh in the WSL(windows subsystem for linux),run:
$ browsh --firefox.use-existing --startup-url https://www.bing.com
or $ browsh --firefox.path /usr/bin/firefox
or $ browth --firefox.with-gui
or $ browth
then the shell turn into a browsh-snail with following words:
Waiting for a user-initiated Firefox instance to connect...

It couldn't move on.I am confused.Thanks for any suggestions!

this is my debug.log
# ~/debug.log when running: $ browth --debug
Looking in /home/younger/.config/browsh/config.toml for config.
Attempting to connect to Firefox Marionette
Starting Firefox in headless mode
Using Firefox at: /usr/bin/firefox
Using default profile at: /home/younger/.config/browsh/firefox_profile
FF-CONSOLE: 1553326113384       Marionette      INFO    Listening on port 2828
FF-MRNT: 50:{"applicationType":"gecko","marionetteProtocol":3}
Sending `WebDriver:NewSession` to Firefox Marionette

# ~/debug.log when running:$ browsh --debug --firefox.use-existing
Looking in /home/younger/.config/browsh/config.toml for config.
Starting Browsh CLI client
Webextension not connected. Message not sent: /tty_size,133,36

**Comment by thetechnologyguy** [comment]
I am having the same issue trying to open a website from a python script. Opens fine from command line in python of I call just browse, but not if I define an address.

Browse 1.5
Ubuntu 18.04
Python 2.7. Same in 3.x
Firefox 65

Heeeelp!!

**Comment by LwsBtlr** [comment]
browsh 1.5.0 macOS 10.4.4 and Firefox 66.0.3. Firefox hs no issues connecting to any site *including this one), but browsh simply sits there waiting.

`browsh https://www.brow.sh/ --firefox.path /Applications/Firefox.app/Contents/MacOS/firefox-bin 
` or `browsh https://www.google.com` or `firefox.use-existing` behave the same.

**Comment by Falkgaard** [comment]
Browsh 1.5.0
Firefox 60.4.0
Running Debian Sid, Konsole,  tmux on my home network
Python etc is installed.
w3m works from my terminal and regular Firefox works without issues.

Debug log:

`Looking in /home/falk/.config/browsh/config.toml for config.
Attempting to connect to Firefox Marionette
Starting Firefox in headless mode
Using Firefox at: /usr/bin/firefox
Using default profile at: /home/falk/.config/browsh/firefox_profile
FF-CONSOLE: 1557018229769	Marionette	INFO	Listening on port 2828
FF-MRNT: 50:{"applicationType":"gecko","marionetteProtocol":3}
Sending `WebDriver:NewSession` to Firefox Marionette
FF-MRNT: 523:[1,0,null,{"sessionId":"c68aa06d-5e39-443d-b4c2-04a57eb3a308","capabilities":{"browserName":"firefox","browserVersion":"60.4.0","platformName":"linux","platformVersion":"4.9.0-8-amd64","pageLoadStrategy":"normal","acceptInsecureCerts":false,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"rotatable":false,"moz:accessibilityChecks":false,"moz:headless":true,"moz:processID":28409,"moz:profile":"/home/falk/.config/browsh/firefox_profile","moz:useNonSpecCompliantPointerOrigin":false,"moz:webdriverClick":true}}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 13:[1,1,null,{}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 25:[1,2,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 13:[1,3,null,{}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 13:[1,4,null,{}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 25:[1,5,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 13:[1,6,null,{}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 13:[1,7,null,{}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 25:[1,8,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 13:[1,9,null,{}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,10,null,{}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,11,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,12,null,{}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,13,null,{}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,14,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,15,null,{}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,16,null,{}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,17,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,18,null,{}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,19,null,{}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,20,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,21,null,{}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,22,null,{}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,23,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,24,null,{}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,25,null,{}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,26,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,27,null,{}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,28,null,{}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,29,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,30,null,{}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,31,null,{}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,32,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 14:[1,33,null,{}]
Sending `Addon:Install` to Firefox Marionette
FF-CONSOLE: 1557018229961	addons.xpi	WARN	Add-on undefined is not correctly signed.
FF-CONSOLE: 1557018229962	addons.xpi	WARN	Add-on undefined is not correctly signed.
FF-CONSOLE: 1557018229963	addons.xpi	WARN	Invalid XPI: Error: Cannot find id for addon /tmp/browsh-webext-addon415744279 (resource://gre/modules/addons/XPIInstall.jsm:1544:17) JS Stack trace: loadManifest@XPIInstall.jsm:1544:17
FF-MRNT: 371:[1,34,{"error":"unknown error","message":"Could not install add-on at '/tmp/browsh-webext-addon415744279': UnknownError: ERROR_CORRUPT_FILE: The file appears to be corrupt.","stacktrace":"WebDriverError@chrome://marionette/content/error.js:178:5\nUnknownError@chrome://marionette/content/error.js:529:5\naddon.install@chrome://marionette/content/addon.js:109:11\n"},null]
Starting Browsh CLI client
Webextension not connected. Message not sent: /tty_size,191,52
Sending `Marionette:Quit` to Firefox Marionette
FF-CONSOLE: console.log: Accessibility service init or shutdown observer does not exist.
FF-MRNT: 32:[1,35,null,{"cause":"shutdown"}]
*errors.errorString normal
/home/travis/gopath/src/browsh/interfacer/src/browsh/tty.go:109 (0x84804b)
/home/travis/gopath/src/browsh/interfacer/src/browsh/tty.go:57 (0x847e3e)
/home/travis/gopath/src/browsh/interfacer/src/browsh/tty.go:45 (0x847bf9)
/home/travis/.gimme/versions/go1.10.5.linux.amd64/src/runtime/asm_amd64.s:2361 (0x457b41)
`

**Comment by tombh** [comment]
> FF-CONSOLE: 1557018229961	addons.xpi	WARN	Add-on undefined is not correctly signed.

OMG! We've been caught by the recent Mozilla Addon Certificate expiry issue! See: https://discourse.mozilla.org/t/certificate-issue-causing-add-ons-to-be-disabled-or-fail-to-install/39047

They've released a fix so all Firefox's should automatically start using the new certificates in the next few hours. If you're desperate you can toggle `xpinstall.signatures.required` in `about:config` when running Browsh's Firefox in GUI mode.

**Comment by Falkgaard** [comment]
Cheers!

**Comment by OvermindDL1** [comment]
Still failing in the most recent docker image (does it need updating?).  It just waits on firefox to connect, let it set for a half hour with no change so far.  Attempted the docker image on 3 different systems running 3 very different linux installs.  `debug.log` is:
```text
Looking in /app/.config/browsh/config.toml for config.
Attempting to connect to Firefox Marionette
Starting Firefox in headless mode
Using Firefox at: /app/bin/firefox/firefox
Using default profile at: /app/.config/browsh/firefox_profile
FF-MRNT: 50:{"applicationType":"gecko","marionetteProtocol":3}
Sending `WebDriver:NewSession` to Firefox Marionette
FF-MRNT: 584:[1,0,null,{"sessionId":"1d570974-3894-4218-ba2f-cc33bc59af1f","capabilities":{"browserName":"firefox","browserVersion":"63.0","platformName":"linux","platformVersion":"4.15.0-39-generic","acceptInsecureCerts":false,"pageLoadStrategy":"normal","setWindowRect":true,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"unhandledPromptBehavior":"dismiss and notify","rotatable":false,"moz:accessibilityChecks":false,"moz:headless":true,"moz:processID":19,"moz:profile":"/app/.config/browsh/firefox_profile","moz:useNonSpecCompliantPointerOrigin":false,"moz:webdriverClick":true}}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 25:[1,1,null,{"value":null}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 25:[1,2,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 25:[1,3,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 25:[1,4,null,{"value":null}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 25:[1,5,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 25:[1,6,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 25:[1,7,null,{"value":null}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 25:[1,8,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 25:[1,9,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,10,null,{"value":null}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,11,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,12,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,13,null,{"value":null}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,14,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,15,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,16,null,{"value":null}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,17,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,18,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,19,null,{"value":null}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,20,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,21,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,22,null,{"value":null}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,23,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,24,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,25,null,{"value":null}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,26,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,27,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,28,null,{"value":null}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,29,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,30,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,31,null,{"value":null}]
Sending `WebDriver:ExecuteScript` to Firefox Marionette
FF-MRNT: 26:[1,32,null,{"value":null}]
Sending `Marionette:SetContext` to Firefox Marionette
FF-MRNT: 26:[1,33,null,{"value":null}]
Sending `Addon:Install` to Firefox Marionette
FF-CONSOLE: 1559336270990       addons.xpi-utils        WARN    Add-on undefined is not correctly signed.
FF-CONSOLE: 1559336270990       addons.xpi-utils        WARN    Add-on undefined is not correctly signed.
FF-CONSOLE: 1559336270997       addons.xpi      WARN    Invalid XPI: Error: Cannot find id for addon /tmp/browsh-webext-addon212400051(resource://gre/modules/addons/XPIInstall.jsm:1549:19) JS Stack trace: loadManifest@XPIInstall.jsm:1549:19
FF-MRNT: 346:[1,34,{"error":"unknown error","message":"Could not install add-on: /tmp/browsh-webext-addon212400051: ERROR_CORRUPT_FILE: The file appears to be corrupt.","stacktrace":"WebDriverError@chrome://marionette/content/error.js:178:5\nUnknownError@chrome://marionette/content/error.js:483:5\ninstall@chrome://marionette/content/addon.js:78:13\n"},null]
Starting Browsh CLI client
Webextension not connected. Message not sent: /tty_size,147,29
FF-CONSOLE: console.error: BroadcastService: 
FF-CONSOLE:   receivedBroadcastMessage: handler for
FF-CONSOLE:   remote-settings/monitor_changes
FF-CONSOLE:   threw error:
FF-CONSOLE:   Message: Error: Polling for changes failed: NetworkError when attempting to fetch resource..
FF-CONSOLE:   Stack:
FF-CONSOLE:     remoteSettingsFunction/remoteSettings.pollChanges@resource://services-settings/remote-settings.js:721:13
FF-CONSOLE: 
Sending `Marionette:Quit` to Firefox Marionette
FF-CONSOLE: 1559336307024       Marionette      INFO    Stopped listening on port 2828
FF-MRNT: 32:[1,35,null,{"cause":"shutdown"}]
*errors.errorString normal
/go-home/src/browsh/interfacer/src/browsh/tty.go:109 (0x86804b)
/go-home/src/browsh/interfacer/src/browsh/tty.go:57 (0x867e3e)
/go-home/src/browsh/interfacer/src/browsh/tty.go:45 (0x867bf9)
/go/src/runtime/asm_amd64.s:2337 (0x45a531)
```

**Comment by tombh** [comment]
@OvermindDL1 This should be fixed in the latest release, let me know.

I'm going to close this issue, because it's a bit of a hold-all, not connecting to Firefox can be caused by many things. So it'd be better to try to isolate individual problems to individual issues.
